subsys/pm/device.c

```diff
@@ -387,8 +387,9 @@ int pm_device_driver_init(const struct device *dev,
 	pm->state = PM_DEVICE_STATE_SUSPENDED;
 
 	/* If device will have PM device runtime enabled */
-	if (IS_ENABLED(CONFIG_PM_DEVICE_RUNTIME) &&
-	    atomic_test_bit(&pm->flags, PM_DEVICE_FLAG_RUNTIME_AUTO)) {
+	if (IS_ENABLED(CONFIG_PM_DEVICE_RUNTIME_DEFAULT_ENABLE) ||
+	    (IS_ENABLED(CONFIG_PM_DEVICE_RUNTIME) &&
+	     atomic_test_bit(&pm->flags, PM_DEVICE_FLAG_RUNTIME_AUTO))) {
```

Comment:
Shouldn't it be something more like:
```c
	if (IS_ENABLED(CONFIG_PM_DEVICE_RUNTIME) &&
	    (IS_ENABLED(CONFIG_PM_DEVICE_RUNTIME_DEFAULT_ENABLE) ||
	    atomic_test_bit(&pm->flags, PM_DEVICE_FLAG_RUNTIME_AUTO))) {
		return 0;
	}
```
This way, it is more logically consistent because in the current version, this **if** suggests that `CONFIG_PM_DEVICE_RUNTIME_DEFAULT_ENABLE` can be enabled without `CONFIG_PM_DEVICE_RUNTIME`.

Reply:
The intended logic is: "If CONFIG_PM_DEVICE_RUNTIME_DEFAULT_ENABLE is enabled, then we know CONFIG_PM_DEVICE_RUNTIME is also enabled, and we don't care about the runtime auto flag. If CONFIG_PM_DEVICE_RUNTIME_DEFAULT_ENABLE is not enabled, we need to know if CONFIG_PM_DEVICE_RUNTIME is enabled, and if it is, we care about the flag"

with the reordering you provided, the PM_DEVICE_FLAG_RUNTIME_AUTO flag will be checked even if CONFIG_PM_DEVICE_RUNTIME is not enabled, which is not desired as the flag does nothing, and will skip resuming the device.